akka-actor-typed-tests/src/test/scala/akka/actor/typed/RequestResponseAdapterSpec.scala

```diff
@@ -0,0 +1,108 @@
+  case class GotPong(id: Long) extends ActorProtocol
+  case class PongTimedOut(id: Long) extends ActorProtocol
+
+  // I think this should go directly on the actor context, but to make experimentation easier I put it here
+  implicit class ContextDecorator[T](ctx: akka.actor.typed.scaladsl.ActorContext[T]) {
+    /**
+     * Works like [[akka.actor.typed.scaladsl.ActorContext.spawnAdapter]] but limited to a single request-response
+     * interaction with another typed actor.
+     *
+     * The interaction has a timeout (to avoid resource a resource leak). If
+     * the timeout hits without any response it will be transformed to a message for this actor through the
+     * `failToOwnProtocol` function.
+     *
+     * @param resToOwnProtocol Transforms the response back from the `otherActor` into a message this actor understands
+     *                         can touch immutable state to provide a context for the interaction, an id for example, but
+     *                         must not touch the `ActorContext` as it will not be executed inside of this actor.
+     *
+     * @tparam Req The request protocol, what the other actor accepts
+     * @tparam Res The response protocol, what the other actor wants to send back
+     */
+    def adaptSingle[Req, Res](otherActor: ActorRef[Req], msgFactory: ActorRef[Res] ⇒ Req)(resToOwnProtocol: Try[Res] ⇒ T)(implicit timeout: Timeout, classTag: ClassTag[Res]): Unit = {
+
+      implicit val ex = ctx.system.executionContext
+      implicit val scheduler = ctx.system.scheduler
+
+      (otherActor ? msgFactory)
+        .mapTo[Res]
+        .onComplete(t ⇒ ctx.self ! resToOwnProtocol(t))
+    }
+
+  }
+
+  implicit val settings = TestKitSettings(system)
+
+  "Single request response interaction adapting" should {
+
+    "not be boilerplatey and work" in {
+
+      val otherActor = start(Actor.immutable[OtherActorProtocol] { (ctx, msg) ⇒
+        msg match {
+          case OtherActorPing(respondTo) ⇒
+            respondTo ! OtherActorPong
+            Actor.same
+        }
+      })
+
+      val probe = TestProbe[AnyRef]()
+
+      implicit val timeout = Timeout(1.second)
+      val actor = start(Actor.immutable[ActorProtocol] { (ctx, msg) ⇒
+
+        msg match {
+          case TriggerPing(id) ⇒
+            // this is it
+            // FIXME naming this is really hard: ideas `requestResponse`, `interact`, `query`, `ask`
+            // FIXME can it be expressed in an even shorter way? (I don't see how but maybe I'm missing something)
+            // note how it allows for capturing context with the id (also dangerous because could be abused closing
+            // over ctx but no idea how to workaround that, we need a response => our-response function after all
+            ctx.adaptSingle(otherActor, OtherActorPing) {
+              case Success(OtherActorPong) ⇒ GotPong(id)
+              case Failure(_)              ⇒ PongTimedOut(id)
+            }
```

Comment:
Yeah looks pretty good; It looks like ask actually so that name is pretty tempting. I also was thinking that having an ask on ctx. would be good, because let's face it it's common enough

Reply:
I agree, `ctx.ask` is great, because it is about asking when you are inside of an actor, which the safe version of is always ask-mapTo-pipe like this.